compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -632,7 +635,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
             ty::ConstKind::Unevaluated(uv) => {
                 let instance = self.resolve(uv.def, uv.substs)?;
-                Ok(self.eval_to_allocation(GlobalId { instance, promoted: uv.promoted })?.into())
+                Ok(self.eval_to_allocation(GlobalId { instance, promoted: None })?.into())
```

Comment:
Why does this make sense? It might be a promoted after all...

Reply:
Also can't this call `uneval_to_op` now?

Reply:
Can you explain why we would want to do that? It's doesn't make sense to me why we should first create a `ConstValue` when we only only want an `Operand` (we would call `eval_to_allocation` for creating the `ConstValue` anyway).